icu.lua

```diff
@@ -19,6 +19,13 @@ project "icu"
     "NoPCH",
   }
 
+  defines {
+    -- Limit the number of conversions supported to only those necessary: ASCII, Latin-1, and
+    -- UTF-8/UTF-16.  See ucnv_bld.cpp for a list of encodings available with each #define
+    "UCONFIG_NO_LEGACY_CONVERSION",
```

Comment:
Do these need to be exposed out at all or are they only contained in cpp files?

Reply:
They do leak out of the headers in a few places: https://github.com/Esri/icu/blob/master/icu4c/source/i18n/csr2022.h#L70

Granted, it doesn't look like we're using any of those ☝️.  There's also this strange practice where they explicitly `#define` it to 0 if [not already defined](https://github.com/Esri/icu/blob/master/icu4c/source/common/unicode/uconfig.h#L295), which isn't something I've seen before.  Again, I don't _think_ it'd get us into too much trouble if dependencies don't inherit these `#defines`, though.

Reply:
So long as they don't leak out, looks good to me. I think they define it to 0 because I've heard that you can hit some issues if you try to check if a define is equal to something but if it is never defined, weird things happen. A lot of ifdef checks from older projects first check if something is defined before checking to see what the value equals so maybe that is why they went with that approach.